cpp/tests/hashing/murmurhash3_x64_128_test.cpp

```diff
@@ -0,0 +1,110 @@
+    {-1, -1, 0, 2, 22, 0, 11, 12, 116, 32, 0, 42, 7, 62, 1, -22, 0, 0},
+    {1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0}};
+  auto col2 = cudf::test::fixed_width_column_wrapper<T, int32_t>{
+    {-1, -1, 0, 2, 22, 1, 11, 12, 116, 32, 0, 42, 7, 62, 1, -22, 1, -22},
+    {1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0}};
+
+  auto output1 = cudf::hashing::murmurhash3_x64_128(cudf::table_view({col1}));
+  auto output2 = cudf::hashing::murmurhash3_x64_128(cudf::table_view({col2}));
+  CUDF_TEST_EXPECT_TABLES_EQUAL(output1->view(), output2->view());
+
+  output1 = cudf::hashing::murmurhash3_x64_128(cudf::table_view({col1}), 7);
+  output2 = cudf::hashing::murmurhash3_x64_128(cudf::table_view({col2}), 7);
+  CUDF_TEST_EXPECT_TABLES_EQUAL(output1->view(), output2->view());
+}
+
+class HashMurmur64Test : public cudf::test::BaseFixture {};
+
+TEST_F(HashMurmur64Test, StringType)
+{
+  auto col1 = cudf::test::strings_column_wrapper(
+    {"The",
+     "quick",
+     "brown fox",
+     "jumps over the lazy dog.",
+     "I am Jack's complete lack of null value",
+     "A very long (greater than 128 bytes/char string) to test a a very long string",
+     "Some multi-byte characters here: ééé",
+     "ééé",
+     "ééé ééé",
+     "ééé ééé ééé ééé",
+     "",
+     "!@#$%^&*(())",
+     "0123456789",
+     "{}|:<>?,./;[]=-"});
+
+  auto output = cudf::hashing::murmurhash3_x64_128(cudf::table_view({col1}));
+  // these were generated using the CPU compiled
+  // https://github.com/aappleby/smhasher/blob/master/src/MurmurHash3.cpp
+  auto expected = cudf::test::fixed_width_column_wrapper<uint64_t>({3481043174314896794ul,
+                                                                    1981901315483788749ul,
+                                                                    1418748153263580713ul,
+                                                                    11224732510765974842ul,
+                                                                    10813495276579975748ul,
+                                                                    3654904410285196488ul,
+                                                                    7289234017606107350ul,
+                                                                    225672801045596944ul,
+                                                                    14927688838032769435ul,
+                                                                    7513581995808204968ul,
+                                                                    0ul,
+                                                                    14163495587303857889ul,
+                                                                    4581940570640870180ul,
+                                                                    18164432652839101653ul});
+
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(output->view().column(0), expected);
+
+  output   = cudf::hashing::murmurhash3_x64_128(cudf::table_view({col1}), 7);
```

Comment:
It took me a while to see the difference was that this was called with a seed. Maybe we can write `auto seed = 7` or add a comment to make that obvious.